nexus/src/app/mod.rs

```diff
@@ -58,6 +61,44 @@ pub(crate) const MAX_NICS_PER_INSTANCE: usize = 8;
 // TODO-completness: Support multiple external IPs
 pub(crate) const MAX_EXTERNAL_IPS_PER_INSTANCE: usize = 1;
 
+pub(crate) struct ExternalServers {
+    config: dropshot::ConfigDropshot,
+    https_port: u16,
+    http: Option<DropshotServer>,
+    https: Option<DropshotServer>,
+}
+
+impl ExternalServers {
+    pub fn new(config: dropshot::ConfigDropshot, https_port: u16) -> Self {
+        Self { config, https_port, http: None, https: None }
+    }
+
+    pub fn set_http(&mut self, http: DropshotServer) {
+        self.http = Some(http);
+    }
```

Comment:
How about adding a `set_https` here for symmetry and using it?

Reply:
Sure; done in b339ed90d9b36b135d32e7771fb6c49f22a08394